core/lib/dal/src/blocks_web3_dal.rs

```diff
@@ -458,7 +458,17 @@ impl BlocksWeb3Dal<'_, '_> {
     pub async fn get_traces_for_miniblock(
         &mut self,
         block_number: MiniblockNumber,
-    ) -> sqlx::Result<Vec<Call>> {
+    ) -> anyhow::Result<Vec<Call>> {
+        let protocol_version = self
+            .storage
+            .blocks_dal()
+            .get_miniblock_protocol_version_id(block_number)
+            .await?;
```

Comment:
you should add unwrap_or_else ProtocolVersionId::last_potentially_undefined

Reply:
but here the error is from DB, shouldn't I propagate it?

Reply:
though for the `if` below, sure